src/viam/sdk/common/exception.hpp

```diff
@@ -0,0 +1,68 @@
+#pragma once
+#include <grpc/status.h>
+#include <grpcpp/impl/codegen/status.h>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+
+namespace viam {
+namespace sdk {
+
+enum class ViamErrorCode : uint8_t {
+    // Used to indicate no error
+    ok = 0,
+    generic = 1,
+    permission_denied = 2,
+    duplicate_resource = 3,
+    unimplemented = 4,
+    validation = 5,
+    connection = 6
+};
+
+class ViamException : public std::runtime_error {
+   public:
+    explicit ViamException(const std::string& what = "unknown");
+
+    explicit ViamException(const std::string& type,
+                           const ViamErrorCode& code,
+                           const std::string& what);
+
+    explicit ViamException(const ::grpc::Status& status_code);
+
+    static ViamException from_viam_error_code(ViamErrorCode code);
+
+    virtual ~ViamException();
+
+    ViamErrorCode get_error_code() const noexcept;
+
+   private:
+    ViamErrorCode _error_code;
```

Comment:
(q) why did you go with a `_` prefix? Elsewhere in the codebase we've been using a suffix pretty consistently.

Reply:
Because I accidentally got confused with python. No reason other than foolishness